data/settings.js

```diff
@@ -515,123 +515,290 @@ depending on the authenticated user).`
+the default script visible in ManageSieve. See
+'[[link,sieve_visible_default_script]]'`
 	},
 
-	sieve_after: {
-		tags: [ 'sieve' ],
+	sieve_script_type: {
+		tags: [ 'sieve-storage' ],
 		plugin: 'sieve',
+		default: 'personal',
+		seealso: [ '[[link,sieve_storage_type]]' ],
 		values: setting_types.STRING,
 		text: `
-This setting can be specified multiple times by adding a number after the
-setting name, such as \`sieve_after2\` and so on.
+The type of the configured Sieve script storage. See
+[[link,sieve_storage_type]].`
+	},
+
+	sieve_script_cause: {
+		tags: [ 'sieve-storage' ],
+		plugin: 'sieve',
+		default: 'delivery',
+		seealso: [ '[[link,sieve_storage]]' ],
+		values: setting_types.STRLIST,
+		text: `
+The causes for executing Sieve scripts from this storage. This is currently only
+relevant for the IMAPSieve plugin. For standard Sieve execution at message
+delivery the cause is "delivery".`
+	},
 
-[[link,sieve_location,Location]] of scripts that need to be executed after
-the user's personal script.
+	sieve_script_precedence: {
+		tags: [ 'sieve-storage' ],
+		plugin: 'sieve',
+		seealso: [ '[[link,sieve_storage]]' ],
+		values: setting_types.UINT,
+		text: `
+The precedence of this Sieve storage in the configuration. Normally, script
+storages with matching type and cause are accessed in the order these are
+specified in the configation. This setting can be used to configure an explicit
+order.`
+	},
 
-If a [[link,sieve_file]] location path points to a directory, all
-the Sieve scripts contained therein (with the proper .sieve extension) are
-executed. The order of execution within that directory is determined by the
-file names, using a normal 8bit per-character comparison.
+	sieve_script_driver: {
+		tags: [ 'sieve-storage' ],
+		plugin: 'sieve',
+		default: 'file',
+		seealso: [ '[[link,sieve_storage]]' ],
+		values: setting_types.STRING,
+		text: `
+The Sieve script storage driver to use. See [[link,sieve_storage_driver]].`
+	},
 
-Multiple script locations can be specified by appending an increasing
-number to the setting name.
+	sieve_script_bin_path: {
+		tags: [ 'sieve-storage' ],
+		plugin: 'sieve',
+		default: 'personal',
+		seealso: [ '[[link,sieve_storage]]' ],
+		values: setting_types.STRING,
+		text: `
+Points to the directory where the compiled binaries for this script
+location are stored. This directory is created automatically if possible.
 
-The Sieve scripts found from these locations are added to the script
-execution sequence in the specified order.
+If this option is omitted, the behavior depends on the storage driver.
 
-Reading the numbered [[setting,sieve_before]] settings stops at the
-first missing setting, so no numbers may be skipped.`
+Don't specify the same directory for different script storages, as this
+will result in undefined behavior. Multiple mail users can share a single
+script directory if the script location is the same and all users share
+the same system credentials (uid, gid).`
```

Comment:
I'm not sure I understand. What does it mean "script location is the same"? If they're all using the same global script? Or even if users have their own private scripts, it somehow stores them in separate directories or something?

Reply:
Location is old terminology. Fixed.